src/resolvers/Mutation/updateUserTag.ts

```diff
@@ -0,0 +1,69 @@
+import { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { errors, requestContext } from "../../libraries";
+import { User, OrganizationTagUser } from "../../models";
+import {
+  USER_NOT_FOUND_ERROR,
+  USER_NOT_AUTHORIZED_ERROR,
+  TAG_NOT_FOUND,
+} from "../../constants";
+
+export const updateUserTag: MutationResolvers["updateUserTag"] = async (
+  _parent,
+  args,
+  context
+) => {
+  const currentUser = await User.findOne({
+    _id: context.userId,
+  }).lean();
+
+  // Checks whether currentUser exists.
+  if (!currentUser) {
+    throw new errors.NotFoundError(
+      requestContext.translate(USER_NOT_FOUND_ERROR.MESSAGE),
+      USER_NOT_FOUND_ERROR.CODE,
+      USER_NOT_FOUND_ERROR.PARAM
+    );
+  }
+
+  // Get the tag object
+  const tag = await OrganizationTagUser.findOne({
+    _id: args.input._id,
+  }).lean();
+
+  if (!tag) {
+    throw new errors.NotFoundError(
+      requestContext.translate(TAG_NOT_FOUND.MESSAGE),
+      TAG_NOT_FOUND.CODE,
+      TAG_NOT_FOUND.PARAM
+    );
+  }
+
+  // Boolean to determine whether user is an admin of organization of the tag folder.
+  const currentUserIsOrganizationAdmin = currentUser.adminFor.some(
+    (organization) => organization.toString() === tag!.organizationId.toString()
+  );
+
+  // Checks whether currentUser can update the tag
+  if (
+    !(currentUser.userType === "SUPERADMIN") &&
+    !currentUserIsOrganizationAdmin
+  ) {
+    throw new errors.UnauthorizedError(
+      requestContext.translate(USER_NOT_AUTHORIZED_ERROR.MESSAGE),
+      USER_NOT_AUTHORIZED_ERROR.CODE,
+      USER_NOT_AUTHORIZED_ERROR.PARAM
+    );
+  }
+
+  // Update the title of the tag folder
+  const updatedTag = await OrganizationTagUser.findOneAndUpdate(
+    {
+      _id: args.input._id,
+    },
+    {
+      name: args.input.name,
+    }
+  );
+
+  return updatedTag!;
```

Comment:
Just return like this:-
```ts
 return await OrganizationTagUser.findOneAndUpdate(
    {
      _id: args.input._id,
    },
    {
      name: args.input.name,
    },
    {
       new: true
    }
  ).lean();
```
Don't forget to use `lean()`, all the useless bloat that mongoose document has doesn't need to be included in returned object, it would just make the server slower.